src/Weather/Skysight/SkysightAPI.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2016 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "SkysightAPI.hpp"
+#include "SkysightRegions.hpp"
+
+SkysightAPI::~SkysightAPI() {
+}
+
+void
+SkysightAPI::Init(tstring email, tstring password, tstring _region)
+{
+  inited_regions = false;
+
+  region = (_region.empty()) ? "EUROPE" : _region;
+  LoadDefaultRegions();
+}
+
+bool
+SkysightAPI::IsInited()
+{
+  return inited_regions;
+}
+
+void
+SkysightAPI::LoadDefaultRegions()
+{
+  for (auto r = skysight_region_defaults; r->id != nullptr; ++r)
+    regions.insert(std::pair<tstring, tstring>(r->id, r->name));
```

Comment:
Use `emplace()`

Reply:
Compile and run fine when I replace insert to emplace.